---
title: "Whole-Genome Sequencing, EPHI"
output: html_notebook
author: "E Reichert"
date: "`r format(Sys.time(), '%B %d, %Y')`"
---

*Background*

This script reads in the whole-genome sequencing (WGS) data for 23 EPHI samples subject to whole-genome sequencing + controls. These visualizations of WGS coverage around pfhrp2/3 (on Chr8 and Chr13, respectively) are utilized for comparing MIP/WGS calls in our manuscript Feleke et al, which identifies pfhrp2/3 deletion mutants in isolates from 2017-18 in Ethiopia. WGS is visualized as the log10 of the median coverage at each 1kb window.

```{r}
#load libraries
library(readr)
library(ggplot2)
library(dplyr)
library(EnvStats)
library(tidyr)
library(stringr)
library(forcats)
library(viridis)

#IMPORT WGS DATA
wgs_chr8 <- read_csv("WGS_CHR8.csv")
wgs_chr13 <- read_csv("WGS_CHR13.csv")
```

*Figure 1. WGS Coverage of Chr8 surrounding pfhrp2*

```{r}
#reshape data into long format, keep genomic window of interest for visualization
wgs_chr8 <- wgs_chr8 %>% gather(key = Gene, value = coverage, -lab_ID, -median) %>%
  mutate(Gene_num = as.numeric(Gene)) %>%
  filter(Gene_num > 1324700 & Gene_num < 1424000) %>%
  select(lab_ID, median, Gene, coverage)

#reassign 0 coverage values to 0.1 for log10 scaling
wgs_chr8$coverage[wgs_chr8$coverage == 0] <- 0.1

#Create normalized WGS coverage score
#median coverage of 1kb bin/median coverage across genome
hrp2_coverage <- wgs_chr8 %>% mutate(z_coverage = coverage/median) %>% 
  arrange((median)) %>%
  select(lab_ID, Gene, coverage, z_coverage)

#order (desc) by median coverage across whole genome (best to worst coverage)
hrp2_coverage$lab_ID <- fct_inorder(hrp2_coverage$lab_ID)

breaks = c(seq(1273735, 1472735, 2000))

#produce heatmap
#here, we are plotting the log10 of median raw coverage in 1kb bins for each sample
pdf(file = "hrp2_wgs.pdf", width = 11, height = 6)
p <- ggplot(hrp2_coverage, aes(x = Gene, y = lab_ID, fill= log10(coverage))) + 
  geom_tile() +
  scale_fill_viridis(discrete=FALSE, option = "plasma") + theme(axis.text.x = element_text(size = 5, angle = 90), axis.text.y = element_text(size = 5), legend.position = "none") + scale_x_discrete(position = "top", breaks = breaks) + ylab("") + xlab("") + labs(fill = "WGS Coverage Score")
p

dev.off()
```

*Figure 2. WGS Coverage of Chr13 surrounding pfhrp3*

```{r}
#reshape data into long format, keep genomic window of interest for visualization
wgs_chr13 <- wgs_chr13 %>% gather(key = Gene, value = coverage, -lab_ID, -median) %>%
  mutate(Gene_num = as.numeric(Gene)) %>%
  filter(Gene_num >= 2771226 & Gene_num <= 2891226) %>%
  select(lab_ID, median, Gene, coverage)

#reassign 0 coverage values to 0.1 for log10 scaling
wgs_chr13$coverage[wgs_chr13$coverage == 0] <- 0.1

#Create normalized coverage score
#median coverage of 1kb bin/median coverage across genome
hrp3_coverage <- wgs_chr13 %>%
  mutate(z_coverage = coverage/median) %>% arrange((median)) %>%
  select(lab_ID, Gene, coverage, z_coverage)

#order (desc) by median coverage across whole genome (best to worst coverage)
hrp3_coverage$lab_ID <- fct_inorder(hrp3_coverage$lab_ID)

breaks = c(seq(2771226, 2891226, 2000))

#produce heatmap
#here, we are plotting the log10 of median raw coverage in 1kb bins for each sample
pdf(file = "hrp3_wgs.pdf", width = 10, height = 6)
p <- ggplot(hrp3_coverage, aes(x = Gene, y = lab_ID, fill= log10(coverage))) + 
  geom_tile() +
  scale_fill_viridis(discrete=FALSE, option = "plasma") + theme(axis.text.x = element_text(size = 5, angle = 90), axis.text.y = element_text(size = 5), legend.position = "none") + scale_x_discrete(position = "top", breaks = breaks) + ylab("") + xlab("") + labs(fill = "WGS Coverage Score")
p

dev.off()
```

